testsuite/MDAnalysisTests/topology/test_tprparser.py

```diff
@@ -232,58 +185,87 @@ def _test_is_in_topology(name, elements, topology_path, topology_section):
     parser = MDAnalysis.topology.TPRParser.TPRParser(topology_path)
     top = parser.parse()
     for element in elements:
-        assert_(element in getattr(top, topology_section).values,
-                'Interaction type "{}" not found'.format(name))
-
-
-def test_all_bonds():
+        assert element in getattr(top, topology_section).values, \
+            'Interaction type "{}" not found'.format(name)
+
+
+@pytest.mark.parametrize('topology', (
+        TPR510_bonded,
+        TPR2016_bonded
+))
+@pytest.mark.parametrize('bond', (
+        ('BONDS', [(0, 1)]),
+        ('G96BONDS', [(1, 2)]),
+        ('MORSE', [(2, 3)]),
+        ('CUBICBONDS', [(3, 4)]),
+        ('CONNBONDS', [(4, 5)]),
+        ('HARMONIC', [(5, 6)]),
+        ('FENEBONDS', [(6, 7)]),
+        ('RESTRAINTPOT', [(7, 8)]),
+        ('TABBONDS', [(8, 9)]),
+        ('TABBONDSNC', [(9, 10)]),
+        ('CONSTR', [(10, 11)]),
+        ('CONSTRNC', [(11, 12)]),
+))
+def test_all_bonds(topology, bond):
     """Test that all bond types are parsed as expected"""
-    topologies = (TPR510_bonded, TPR2016_bonded)
-
-    bonds = {'BONDS':[(0, 1)], 'G96BONDS':[(1, 2)], 'MORSE':[(2, 3)],
-             'CUBICBONDS':[(3, 4)], 'CONNBONDS':[(4, 5)], 'HARMONIC':[(5, 6)],
-             'FENEBONDS':[(6, 7)], 'RESTRAINTPOT':[(7, 8)],
-             'TABBONDS':[(8, 9)], 'TABBONDSNC':[(9, 10)],
-             'CONSTR':[(10, 11)], 'CONSTRNC':[(11, 12)],}
     bond_type_in_topology = functools.partial(_test_is_in_topology,
                                               topology_section='bonds')
-    for topology in topologies:
-        for bond_type, elements in bonds.items():
-            yield (bond_type_in_topology, bond_type, elements, topology)
-
-
-def test_all_angles():
-    topologies = (TPR510_bonded, TPR2016_bonded)
-    angles = {'ANGLES':[(0, 1, 2)], 'G96ANGLES':[(1, 2, 3)],
-              'CROSS_BOND_BOND':[(2, 3, 4)], 'CROSS_BOND_ANGLE':[(3, 4, 5)],
-              'UREY_BRADLEY':[(4, 5, 6)], 'QANGLES':[(5, 6, 7)],
-              'RESTRANGLES':[(6, 7, 8)], 'TABANGLES':[(7, 8, 9)],}
+    bond_type, elements = bond
+    bond_type_in_topology(bond_type, elements, topology)
+
+
+@pytest.mark.parametrize('topology', (
+    TPR510_bonded,
+    TPR2016_bonded
+))
+@pytest.mark.parametrize('angle', (
+    ('ANGLES',[(0, 1, 2)]),
+    ('G96ANGLES',[(1, 2, 3)]),
+    ('CROSS_BOND_BOND',[(2, 3, 4)]),
+    ('CROSS_BOND_ANGLE',[(3, 4, 5)]),
+    ('UREY_BRADLEY',[(4, 5, 6)]),
+    ('QANGLES',[(5, 6, 7)]),
+    ('RESTRANGLES',[(6, 7, 8)]),
+    ('TABANGLES',[(7, 8, 9)]),
+))
+def test_all_angles(topology, angle):
     angle_type_in_topology = functools.partial(_test_is_in_topology,
```

Comment:
The partial functions were there to make the verbose output more readable, they are not useful anymore.